### PR TITLE
feat: Migrate Market Details page

### PR DESCRIPTION
**Pull request by jaredvu**
Screen Shot 2024-12-18 at 3 45 42 PM

Migrate CurrentMarketDetails page

---

## Views

- New: `<CurrentMarketDetails2>`
  - use abacusTs marketInfo to populate CurrentMarketDetails

- `<InnerPanel>` / `<MobileInnerPanel>`
  -  use testflag to conditionally render CurrentMarketDetails2

## State

- `abacusTs/calculators/markets`
  - move `getMarketEffectiveInitialMarginForMarket` from subaccounts calculator
  - return displayableAsset, displayableTicker, and effectiveInitialMarginFraction as part of MarketInfo

- `abacusTs/selectors/markets`
  - add `selectCurrentMarketInfo`

- `abacusTs/summaryTypes`
  - add displayableAsset, displayableTicker, and effectiveInitialMarginFraction as part of MarketInfo

---

### CODE REVIEW

File: src/views/MarketDetails/CurrentMarketDetails2.tsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+import { BIG_NUMBERS } from '@/lib/numbers';
+import { orEmptyObj } from '@/lib/typeUtils';
+
+import { MarketDetails } from './MarketDetails';
+
+export const CurrentMarketDetails = () => {
+  const stringGetter = useStringGetter();
+  const currentMarketData = useAppSelector(selectCurrentMarketInfo, shallowEqual);
+  const asset = useMetadataServiceAssetFromId(currentMarketData?.ticker);
```
Comment (tyleroooo) [inline src/views/MarketDetails/CurrentMarketDetails2.tsx:25]: why not use the asset data in the state? Is it missing stuff?

**Comment by tyleroooo** [comment]
Everything else looks goooood I think. I've been doing all the replacements in place to make writing/reading the diffs easier. We should get in sync on if we're going to feature flag this or just roll it out. I am inclined to just roll out.

File: src/views/MarketDetails/CurrentMarketDetails2.tsx (hunk 1 above)
Comment (tyleroooo) [reply src/views/MarketDetails/CurrentMarketDetails2.tsx:25]: Maybe we should replace the assets in raw state with the query inside useMetadataService actually.

**Comment by jaredvu** [comment]
> Everything else looks goooood I think. I've been doing all the replacements in place to make writing/reading the diffs easier. We should get in sync on if we're going to feature flag this or just roll it out. I am inclined to just roll out.

Sounds good. I don't mind having partial roll out for some components (this one is quick and easy to clean up + view doesn't have too many lines of code) but if you don't think thats a good idea I can just do in place

**Comment by tyleroooo** [comment]
I vote we just do it in place. For the rest calls there's not much that could go wrong so it should just work without any issues I think.

File: src/abacus-ts/summaryTypes.ts [hunk 2]
```diff
@@ -21,6 +21,9 @@ type ConvertStringToBigNumber<T, K extends SelectStringProperties<T>> = {
 };
 
 export type MarketInfo = IndexerPerpetualMarketResponseObject & {
+  displayableAsset: string;
```
Comment (tyleroooo) [inline src/abacus-ts/summaryTypes.ts:24]: can we add assetId here too?

File: src/abacus-ts/calculators/assets.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  })
+);
+
+export const transformAssetsInfo = (assetsInfo: MetadataServiceInfoResponse | undefined) => {
+  if (assetsInfo == null) {
+    return assetsInfo;
+  }
+
+  // Add id field to each assetInfo
+  return mapValues(assetsInfo, (assetInfo, key) => parseAssetInfo({ ...assetInfo, assetId: key }));
```
Comment (tyleroooo) [inline src/abacus-ts/calculators/assets.ts:25]: the spread here breaks weakmapmemoize, you need to pass the id as a separate argument

File: src/abacus-ts/selectors/assets.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { selectRawAssetsData } from './base';
+
+export const selectAllAssetsInfo = createSelector([selectRawAssetsData], (assets) =>
+  transformAssetsInfo(assets)
+);
+
+export const selectCurrentMarketAssetInfo = createSelector(
+  [getCurrentMarketId, selectAllAssetsInfo],
+  (currentMarket, assets) => {
+    const assetId = getAssetFromMarketId(currentMarket ?? '');
```
Comment (tyleroooo) [inline src/abacus-ts/selectors/assets.ts:17]: if we put assetId on the market then could use selectcurrentmarketinfo.assetId

File: src/abacus-ts/selectors/markets.ts [hunk 5]
```diff
@@ -6,3 +7,8 @@ import { selectRawMarketsData } from './base';
 export const selectAllMarketsInfo = createAppSelector([selectRawMarketsData], (markets) =>
   calculateAllMarkets(markets)
 );
+
+export const selectCurrentMarketInfo = createAppSelector(
+  [selectAllMarketsInfo, (state: RootState) => state.perpetuals.currentMarketId],
```
Comment (tyleroooo) [inline src/abacus-ts/selectors/markets.ts:12]: might as well use the selectCurrentMarketId which I think exists

File: src/lib/assetUtils.ts [hunk 6]
```diff
@@ -102,3 +102,8 @@ export const getDefaultChainIDFromNetworkType = (networkType: NetworkType): stri
   if (networkType === 'cosmos') return 'noble-1';
   return undefined;
 };
+
+export const getAssetDescriptionStringKeys = (assetId: string) => ({
```
Comment (tyleroooo) [inline src/lib/assetUtils.ts:106]: where did this used to be?

Comment (jaredvu) [reply src/lib/assetUtils.ts:106]: it never existed! It was something returned form Abacus in the resources object

**Review by tyleroooo — APPROVED** [review]

**Pull request merged**